src/main/java/com/algorand/algosdk/builder/transaction/KeyRegistrationTransactionBuilder.java

```diff
@@ -221,4 +231,49 @@ public T voteKeyDilution(Long voteKeyDilution) {
         this.voteKeyDilution = BigInteger.valueOf(voteKeyDilution);
         return (T) this;
     }
+
+    /**
+     * Set the stateProofKey value. This is used to identify a state proof
+     * @param stprf The stateProofKey value.
+     * @return This builder.
+     */
+    public T stateProofKey(MerkleVerifier stprf) {
+        this.stateProofKey = stprf;
+        return (T) this;
+    }
+
+    /**
+     * Set the stateProofKey value. This is used to identify a state proof
+     * @param stprf The stateProofKey value.
+     * @return This builder.
+     */
+    public T stateProofKeyByte(byte[] stprf) {
+        this.stateProofKey = new MerkleVerifier(stprf);
+        return (T) this;
+    }
+
+    /**
+     * Set the stateProofKey value. This is used to identify a state proof
+     * @param stprf The stateProofKey value.
+     * @return This builder.
+     */
+    public T stateProofKeyBase64(String stprf) {
+        if(!stprf.isEmpty()){
```

Comment:
I think I need to keep this check here so that stateProofKey can be an optional field by setting it to null. If not set to null, an empty stateProofKey is included in the txn as, `"sprfkey": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=="`